config/locales/en.yml

```diff
@@ -82,6 +82,10 @@ en:
             using_producer_stock_settings_but_count_on_hand_set: "must be blank because using producer stock settings"
             on_demand_but_count_on_hand_set: "must be blank if on demand"
             limited_stock_but_no_count_on_hand: "must be specified because forcing limited stock"
+        spree/product:
+          attributes:
+            variant_unit:
+              blank: "^Unit Size cannot be blank"
```

Comment:
I'm not that happy with this solution, but that's the only one I've found.
Notice the `^` at the beginning of the message that means to not format the error message with the attribute name (here specify as "variant unit")

I wanted to specify an overridden key for the attribute name, to replace in error message "Variant Unit" to "Unit Size" (which is the work used in the form): 
<img width="530" alt="Capture d’écran 2022-05-24 à 16 56 02" src="https://user-images.githubusercontent.com/296452/170066721-f42f214a-ffcf-49d6-8df0-ebbc22c07fc8.png">

But i didn't manage to.